linkerd/system/src/linux.rs

```diff
@@ -39,8 +42,8 @@ pub fn open_fds(pid: pid_t) -> io::Result<u64> {
 }
 
 pub fn max_fds() -> io::Result<Option<u64>> {
-    let limit = pid::limits_self()?.max_open_files;
-    let max_fds = limit.soft.or(limit.hard).map(|max| max as u64);
+    let limit1 = Process::myself().unwrap().limits().unwrap().max_open_files;
```

Comment:
Since we are returning a `Result` anyway, we probably should just unwrap everything but handle the errors.
```suggestion
    let limit1 = Process::myself()?.limits()?.max_open_files;
```

Reply:
Have modified this to handle error using match. We cannot use '?' here because the error return type here is Error whereas the Process::myself() is ProcError.